lib/http/timeout/per_operation.rb

```diff
@@ -20,40 +17,22 @@ def initialize(*args)
         @connect_timeout = options.fetch(:connect_timeout, CONNECT_TIMEOUT)
       end
 
-      def connect(_, host, port)
-        # https://github.com/celluloid/celluloid-io/blob/master/lib/celluloid/io/tcp_socket.rb
-        begin
-          addr = Resolv::IPv4.create(host)
-        rescue ArgumentError
-        end
-
-        # Guess it's not IPv4! Is it IPv6?
-        begin
-          addr ||= Resolv::IPv6.create(host)
-        rescue ArgumentError
-        end
-
-        unless addr
-          addr = resolve_address(host)
-          fail Resolv::ResolvError, "DNS result has no information for #{host}" unless addr
+      def connect(socket_class, host, port)
+        ::Timeout.timeout(connect_timeout, TimeoutError) do
+          @socket = socket_class.open(host, port)
         end
+      end
 
-        case addr
-        when Resolv::IPv4
-          family = Socket::AF_INET
-        when Resolv::IPv6
-          family = Socket::AF_INET6
-        else fail ArgumentError, "unsupported address class: #{addr.class}"
+      def connect_ssl
+        socket.connect_nonblock
+      rescue IO::WaitReadable
```

Comment:
I guess I missed this on the last pass, but as the SSL/TLS handshake makes many roundtrips, I believe it's possible for IO::WaitWritable to be raised here too.